lib/modules/dosomething/dosomething_api/plugins/restful/campaigns/1.0/DoSomethingCampaignsResource.class.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * @file
+ * Contains DoSomethingCampaignsResource.
+ */
+class DoSomethingCampaignsResource extends RestfulEntityBaseNode {
+
+  /**
+   * Overrides RestfulEntityBaseNode::publicFieldsInfo().
+   */
+  public function publicFieldsInfo() {
+    $public_fields = parent::publicFieldsInfo();
+
+    $public_fields['type'] = array(
+      'property' => 'field_campaign_type',
+    );
+
+    $public_fields['status'] = array(
+      'property' => 'field_campaign_status',
+    );
+
+    $public_fields['call_to_action'] = array(
+      'property' => 'field_call_to_action',
+    );
+
+    $public_fields['on_mobile_app'] = array(
+      'property' => 'field_on_mobile_app',
+    );
+
+    // This is so wtf.
+    // If I name it as 'cover_image' it doesn't work.
+    // If I name it as just 'image' it works.
```

Comment:
Yeah good question, that was one of @aaronschachter's open questions. I'll look into it.